### PR TITLE
Remove payloads

### PR DESCRIPTION
**Pull request by viktorprogger**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | ❌

After some code reviewing I've understood: the library uses only `MessageInterface` inside itself. `PayloadInterface` was designed to make message creating more comfortable. In this PR I've removed payloads and manual metadata managing. I've created message behaviors to make message configuration comfortable.

Usage before:
```php
class DownloadJob implements PayloadInterface, DelayablePayloadInterface
{
    public const NAME = 'file-download';
    public string $url;	
    public string $fileName;
    	
    public function __construct(string $url, string $fileName)	
    {	
        $this->url = $url;	
        $this->fileName = $fileName;	
    }	
    	
    public function getName(): string	
    {	
        return self::NAME;	
    }	
    public function getData(): array	
    {	
        return [	
            'destinationFile' => $this->fileName,	
            'url' => $this->url	
        ];	
    }	


    public function getMeta(): array
    {
        return [];
    }	

    public function getDelay(): int	
    {	
        return 5 * 60;	
    }
}

$payload = new DownloadJob($url, $filename);
$queue->push($payload);
```

After:

```php
$data = [
    'url' => $url,
    'destinationFile' => $filename,
];
$message = new Message('file-download', $data);
$message->attachBehavior(new DelayBehavior(5 * 60));
$queue->push($message);
```

You can also implement `MessageInterface` into smthng like this to use typehints in constructor:

```php
class Message extends AbstractMessage
{
    public const NAME = 'file-download';

    private string $url;
    private string $fileName;

    public function __construct(string $url, string $fileName)
    {
        $this->url = $url;	
        $this->fileName = $fileName;
    }

    public function getName(): string
    {
        return self::NAME;
    }

    public function getData(): array
    {
        return [
            'destinationFile' => $this->fileName,	
            'url' => $this->url	
        ];
    }
}
```

`Queue::jobRetry()` is removed to avoid adopting it in anticipation of #64

### CODE REVIEW

**Comment by viktorprogger** [comment]
Will add some more documentation before review.

File: src/Driver/BehaviorCheckerInterface.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Queue\Driver;
+
+use Yiisoft\Yii\Queue\Exception\BehaviorNotSupportedException;
+
+interface BehaviorCheckerInterface
```
Comment (samdark) [inline src/Driver/BehaviorCheckerInterface.php:9]: Is there going to ever be another implementation?

File: src/Message/Behaviors/PriorityBehaviorInterface.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Queue\Message\Behaviors;
+
+interface PriorityBehaviorInterface
```
Comment (samdark) [inline src/Message/Behaviors/PriorityBehaviorInterface.php:7]: Is there going to be another implementation?

File: src/Message/Behaviors/DelayBehaviorInterface.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Queue\Message\Behaviors;
+
+interface DelayBehaviorInterface
```
Comment (samdark) [inline src/Message/Behaviors/DelayBehaviorInterface.php:7]: Is there going to be another implementation?

**Comment by samdark** [comment]
👍

**Pull request merged**